numpy/core/src/umath/_scaled_float_dtype.c

```diff
@@ -194,13 +194,76 @@ sfloat_get_scaling(PyArray_SFloatDescr *self, PyObject *NPY_UNUSED(args))
 }
 
 
+static PyObject *
+sfloat___reduce__(PyArray_SFloatDescr *self)
+{
+    PyObject *ret, *mod, *func, *obj, *state;
+
+    ret = PyTuple_New(3);
+    if (ret == NULL) {
+        return NULL;
+    }
+
+    mod = PyImport_ImportModule("numpy.core._multiarray_umath");
+    if (mod == NULL) {
+        Py_DECREF(ret);
+        return NULL;
+    }
+
+    func = PyObject_GetAttrString(mod, "_get_sfloat_dtype");
+    Py_DECREF(mod);
+    if (func == NULL) {
+        Py_DECREF(ret);
+        return NULL;
+    }
+
+    obj = PyObject_CallNoArgs(func);
+    Py_DECREF(func);
+    if (obj == NULL) {
+        Py_DECREF(ret);
+        return NULL;
+    }
+
+    PyTuple_SET_ITEM(ret, 0, obj);
+
+    PyTuple_SET_ITEM(ret, 1, PyTuple_New(0));
+
+    state = PyTuple_New(1);
+
+    PyTuple_SET_ITEM(state, 0, PyFloat_FromDouble(self->scaling));
+
+    PyTuple_SET_ITEM(ret, 2, state);
+
+    return ret;
```

Comment:
```suggestion
    return Py_BuildValue("(O(d))", Py_TYPE(self), self->scaling);
```
Happy to monkey-patch numpy in the test, but then lets keep this simple.  (I don't generally love `Py_BuildValue`, but it can be nice and short.)